### ISSUE TITLE
SMTP settings from env variables with docker-compose on install

### ISSUE THREAD

**Issue by tswfi** [PrestaShop/docker#254]
My dev environment is built on top of presta docker image and mysql and mailcatcher in a nice little docker-compose.yml file.

when the environment is started for the first time everything else is quite good but smtp settings cannot be set from env variables.

Please add few new env settings for setting the smtp settings for the newly installed shop:
![image](https://user-images.githubusercontent.com/1352132/107378318-8623f500-6af4-11eb-9a20-325723be5c7e.png)

this would probably be useful for some production environments also.

this requires adding the settings to the cli installer also.

**Comment by PierreRambaud** [comment]
Hey, you make me think, this is something we should be able to add in the installer and be customizable by environment variable. But must be done in the Core project because not sure it's a good idea to manipulate the database here :/

**Comment by tswfi** [comment]
Yup, I wasn't sure which project this issue should be first started :)

**Comment by PierreRambaud** [comment]
Why not a Symfony COmmand to change values in the configuration table?

**Comment by PululuK** [comment]
in install process  :  CLI (news params) + WEB (html form) ?

**Comment by PierreRambaud** [comment]
I think it's something that can be called after the installation to populate in CLI. Not sure it's a good idea to add something in WEB because options are already plugged to many forms inside the BO

**Comment by tswfi** [comment]
@PululuK I don't think the web installer should ask these, it should be easy and straightforward for beginners. This is kinda expert thing..

@PierreRambaud yeah, all the settings should have commands ;) but how to trigger the command in docker for the first startup?

**Comment by PierreRambaud** [comment]
@tswfi docker image automatically install PrestaShop, so it can be done just after the installation :)

**Comment by tswfi** [comment]
@PierreRambaud is there a "post install hook" that can trigger changing the shop settings? (and a cli command for setting the smtp settings? ;)

Of course one can write extra scripts to change the values, but it would be nice to have something built in for getting developers up and running from day one without fiddling with smtp settings. (and a must for unattended installation if one wants to run a saas service)

This also touches the testing aspects of the shop. I haven't checked how the emails are currently tested but mailcather could solve that too. Test cases could check from mailcather that the email was correctly sent and contains the correct data. (or is there a solution for testing these already?)

**Comment by PierreRambaud** [comment]
> @PierreRambaud is there a "post install hook" that can trigger changing the shop settings? (and a cli command for setting the smtp settings? ;)
> 
> Of course one can write extra scripts to change the values, but it would be nice to have something built in for getting developers up and running from day one without fiddling with smtp settings. (and a must for unattended installation if one wants to run a saas service)
> 
> This also touches the testing aspects of the shop. I haven't checked how the emails are currently tested but mailcather could solve that too. Test cases could check from mailcather that the email was correctly sent and contains the correct data. (or is there a solution for testing these already?)

There is no post install hook in PrestaShop, and no CLI command.
But, with a command in CLI, we should be able to use the post install hook (in Docker build) to do it :)

Some of us are using mailcatcher, but have to manually configure it in the BO since there is nothing right now :(
